### PR TITLE
Read tenant from JWT's "tid" claim

### PR DESCRIPTION
**Pull request by sophokles73**
This is for #3441

As a follow-up on
https://github.com/eclipse-hono/hono/pull/3451#discussion_r1064156006 the JwtAuthHandler has been changed to read the tenant Id from a JWT's "tid" claim instead of the "iss" claim. This allows for the "iss" claim to be used for determining where the public key to use for validating the JWS signature should be looked up from, e.g. the tenant configuration or device credentials.

### CODE REVIEW

**Comment by sophokles73** [comment]
@julian-sotec would you mind taking a look if this is still compatible with your use case?

**Comment by julian-sotec** [comment]
> @julian-sotec would you mind taking a look if this is still compatible with your use case?

@sophokles73 since our use case involves the the client identifier to extract the tenantId and `[parseCredentialsFromClientIdentifier](https://github.com/eclipse-hono/hono/blob/ea5ddd863c7ea0c5d874fe023d10b00fb91e46a9/adapters/mqtt-base/src/main/java/org/eclipse/hono/adapter/mqtt/JwtAuthHandler.java#L149)`  method still does this - I don't see any compatibility issues

**Comment by mattkaem** [comment]
> @julian-sotec would you mind taking a look if this is still compatible with your use case?

Please let me also take a look at the changes since I wrote this feature :)

**Comment by sophokles73** [comment]
@mattkaem did you take a look?

File: site/documentation/content/concepts/device-identity.md [hunk 1]
```diff
@@ -150,37 +150,42 @@ the client certificate's *subject DN* and invokes the Credentials API's
...
+[JSON Web Token](https://www.rfc-editor.org/rfc/rfc7519) (JWT) based mechanism.
+In this case, the protocol adapter tries to validate the token presented by the device using a public key on record.
+
+During connection establishment the device presents a client identifier, a username and a password in its
+MQTT CONNECT packet to the protocol adapter. The value of the username field is ignored. The password field is
+expected to contain a valid [JSON Web Signature](https://www.rfc-editor.org/rfc/rfc7515) (JWS) structure that
+contains a JWT in its payload.
+
+The JOSE header MUST contain the *typ* property with value `JWT` and MUST contain the *alg* property indicating the
+algorithm that has been used to create the JWS signature. `RS256`, `ES256` and their respective stronger
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply site/documentation/content/concepts/device-identity.md:163]: my bad, I will add it again ...

File: site/documentation/content/concepts/device-identity.md [hunk 2]
```diff
@@ -150,37 +150,42 @@ the client certificate's *subject DN* and invokes the Credentials API's
...
+The MQTT protocol adapter supports authentication of devices with a
+[JSON Web Token](https://www.rfc-editor.org/rfc/rfc7519) (JWT) based mechanism.
+In this case, the protocol adapter tries to validate the token presented by the device using a public key on record.
+
+During connection establishment the device presents a client identifier, a username and a password in its
+MQTT CONNECT packet to the protocol adapter. The value of the username field is ignored. The password field is
+expected to contain a valid [JSON Web Signature](https://www.rfc-editor.org/rfc/rfc7515) (JWS) structure that
+contains a JWT in its payload.
+
+The JOSE header MUST contain the *typ* property with value `JWT` and MUST contain the *alg* property indicating the
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply site/documentation/content/concepts/device-identity.md:162]: That is what it is called in the JWS spec. But I guess we can simply replace it with
> The JWT's header MUST ....

WDYT?

[the remark this answers is not in the document]
Comment (mattkaem) [reply site/documentation/content/concepts/device-identity.md:162]: I like it, this way it is way less confusing.

**Comment by mattkaem** [comment]
> @mattkaem did you take a look?

yes, it looks very good now 👍

**Pull request merged**